### PR TITLE
CORS refactoring.

### PR DESCRIPTION
**Pull request by dkorolev**
After https://github.com/C5T/Current/pull/805, just one commit: https://github.com/C5T/Current/commit/b5a9a9ba0eda24b57e37695d66be5a5a558fe924

I took the liberty to flip the order of headers and content type, as headers are clearly more important :-)

CC @grixa

### CODE REVIEW

File: blocks/http/response.h [hunk 1]
```diff
@@ -248,9 +238,25 @@ struct Response {
     return *this;
   }
 
-  void RespondViaHTTP(Request r) const {
+  Response& NoCORS() {
```
Comment (sompylasar) [inline blocks/http/response.h]: NoCORS -> DisableCORS

File: blocks/http/response.h [hunk 2]
```diff
@@ -248,9 +238,25 @@ struct Response {
...
+    cors_header = false;
+    return *this;
+  }
+
+  Response& ReinsertCORS() {
```
Comment (sompylasar) [inline blocks/http/response.h]: ReinsertCORS -> EnableCORS

**Comment by dkorolev** [comment]
PTAL @ 9407286c

**Comment by dkorolev** [comment]
Heads up: I'll probably need this over the weekend (well, Sat morning and Sun, as Sat afternoon @grixa and me won't be coding :slightly_smiling_face:), and will merge it myself Sat morning unless you guys have more comments.

File: bricks/net/http/headers/headers.h [hunk 3]
```diff
@@ -274,10 +278,8 @@ struct Headers final {
     }
   }
 
-  // Can `Header& operator[]`. Creates the header if it does not exist. Its `.header` is const, `.value`
-  // mutable.
-  // Capitalization-wise, treats various spellings of the same header as one header, retains the name first
-  // seen.
+  // Can `Header& operator[]`. Creates the header if it does not exist. Its `.header` is const, `.value` mutable.
```
Comment (sompylasar) [inline bricks/net/http/headers/headers.h]: Can what?

Comment (dkorolev) [reply bricks/net/http/headers/headers.h]: ad733d51

**Pull request merged**